reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Retry.kt

```diff
@@ -0,0 +1,34 @@
+package com.badoo.reaktive.completable
+
+import com.badoo.reaktive.base.operator.Retry
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+import kotlin.reflect.KClass
+
+fun Completable.retry(predicate: (Int, Throwable) -> Boolean = { _, _ -> true }): Completable =
+    completable { emitter ->
+        val retry = Retry(emitter, predicate)
+
+        val disposableWrapper = DisposableWrapper()
+        emitter.setDisposable(disposableWrapper)
+
+        subscribeSafe(
+            object : CompletableObserver, CompletableCallbacks by emitter {
+
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onError(error: Throwable) {
+                    retry.onError(error) { this@retry.subscribeSafe(this) }
+                }
+            }
+        )
+    }
+
+fun Completable.retry(times: Int): Completable =
+    retry { attempt, _ -> attempt < times }
+
+fun Completable.retry(throwableType: KClass<out Throwable>): Completable =
```

Comment:
Maybe also add inline reified variant?

Reply:
![Screenshot 2019-09-22 at 20 47 08](https://user-images.githubusercontent.com/9081555/65393577-59920080-dd7a-11e9-9db2-f1cd3bc06a80.png)

Reply:
It clashes with base implementation.

Explicit statement below won't help too.
```kotlin
fun <T> Single<T>.retry(): Single<T> =
    retry { _, _ -> true }
```

Reply:
You have to specify the type explicitly if there are more operators chained after `retry()`. This is fine. And I don't understand what exactly clashes. All the operators have different signatures.

Reply:
No, clash between `retry<Throwable>()` and `retry()` of `retry( {Int, Throwable} -> Boolean = { _, _ -> true} )`. Second one has default parameter and can be called like `retry(), that us why clash occurs.